usr/share/rear/prep/default/320_include_uefi_env.sh

```diff
@@ -47,13 +47,21 @@ if grep -qw efivars /proc/mounts; then
     SYSFS_DIR_EFI_VARS=/sys/firmware/efi/efivars
 fi
 
-# next step, is checking /boot/efi directory (we need it)
-if [[ ! -d /boot/efi ]]; then
-    return    # must be mounted
+# next step, is case-sensitive checking /boot for case-insensitive /efi directory (we need it)
+if [[ -n `find /boot -maxdepth 1 -iname efi -type d` ]]; then
+    return    # not found
 fi
 
 # next step, check filesystem partition type (vfat?)
+local efi_mount_point=""
 UEFI_FS_TYPE=$(awk '/\/boot\/efi/ { print $3 }' /proc/mounts)
+# if not mounted at /boot/efi, try /boot itself
+if [[ -z "$UEFI_FS_TYPE" ]]; then
+    UEFI_FS_TYPE=$(awk '/\/boot/ { print $3 }' /proc/mounts)
+    [[ -z "$UEFI_FS_TYHPE" ]] && efi_mount_point='/\/boot/'
```

Comment:
I understand what you try to fix with this commit and I'm OK with that (even I still think that /boot should not be on vfat).

What I don't understand is following:
Your code:
```
if [[ -z "$UEFI_FS_TYPE" ]]; then
      UEFI_FS_TYPE=$(awk '/\/boot/ { print $3 }' /proc/mounts)
     [[ -z "$UEFI_FS_TYPE" ]] && efi_mount_point='/\/boot/'
  else
      efi_mount_point='/\/boot\/efi/'
fi
```

With ` [[ -z "$UEFI_FS_TYPE" ]] && efi_mount_point='/\/boot/'` you try to say "If $UEFI_FS_TYPE is empty set  efi_mount_point to _/boot_

But later code:
```
if [[ "$UEFI_FS_TYPE" != "vfat" ]]; then
    return
fi
```

Meaning if $UEFI_FS_TYPE does not contain string "vfat" this is not and UEFI enabled system.
So if your condition `[[ -z "$UEFI_FS_TYPE" ]]` is TRUE, there is no reason for setting _efi_mount_point_ as `if [[ "$UEFI_FS_TYPE" != "vfat" ]]` condition will be TRUE as well.